packages/jsii-pacmak/lib/targets/dotnet.ts

```diff
@@ -126,44 +128,677 @@ export default class Dotnet extends Target {
     }
 }
 
-// ##################
-// # CODE GENERATOR #
-// ##################
+// #####################
+// # CODE GENERATOR V2 #
+// #####################
+class DotNetGenerator extends Generator {
+
+    // The path of the original jsii input model.
+    private jsiiFilePath: string;
+
+    // The dependency tree for the current jsii input model.
+    // This is later used to output the csproj
+    private dependencies: Map<string, DotNetDependency> = new Map<string, DotNetDependency>();
 
-class DotNetGenerator implements IGenerator {
-    private jsiiFile: string;
+    constructor(options = new GeneratorOptions()) {
+        super(options);
 
-    public generate(): void {
-        // The DotNet generator does not currently support in-memory generation.
-        // Support can be added relatively easily if necessary.
+        // Override the openBlock to get a correct C# looking code block with the curly brace after the line
+        this.code.openBlock = function(s) {
+            this.line(s);
+            this.open('{');
+        };
     }
 
     public async load(packageRoot: string) {
```

Comment:
I see you are not adding return types to your functions. Please do that. It offers us better type-safety checks and intellisense.